sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesId.cs

```diff
@@ -0,0 +1,157 @@
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 keys.
+        /// </remarks>
+        public TimeSeriesId(object key1, object key2, object key3)
+        {
+            _key1 = (true, key1);
+            _key2 = (true, key2);
+            _key3 = (true, key3);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 1 string key.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 1 string key.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1)
+        {
+            _key1 = (true, stringKey1);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 2 string keys.
+        /// </summary>
+        /// <param name="stringKey1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 2 string keys.
+        /// </remarks>
+        public TimeSeriesId(string stringKey1, string timeSeriesIdStringProp2)
+        {
+            _key1 = (true, stringKey1);
+            _key2 = (true, timeSeriesIdStringProp2);
+        }
+
+        /// <summary>
+        /// Creates a new Time Series Id with 3 string keys.
+        /// </summary>
+        /// <param name="timeSeriesIdStringProp1">The first key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp2">The second key that identifies a Time Series instance.</param>
+        /// <param name="timeSeriesIdStringProp3">The third key that identifies a Time Series instance.</param>
+        /// <remarks>
+        /// Use this constructor if the Time Series Id chosen at environment creation time
+        /// is composed of 3 string keys.
+        /// </remarks>
+        public TimeSeriesId(string timeSeriesIdStringProp1, string timeSeriesIdStringProp2, string timeSeriesIdStringProp3)
+        {
+            _key1 = (true, timeSeriesIdStringProp1);
+            _key2 = (true, timeSeriesIdStringProp2);
+            _key3 = (true, timeSeriesIdStringProp3);
+        }
+
+        /// <summary>
+        /// Builds an array to represent a single Time Series Id in order to use it when making calls against the Time
+        /// Series Insights client library.
+        /// </summary>
+        /// <returns>An array representing a single Time Series Id.</returns>
+        public object[] ToArray()
+        {
+            if (_key3.Item1)
+            {
+                return new object[] { _key1.Item2, _key2.Item2, _key3.Item2 };
```

Comment:
Does it need a empty check for other keys?

Reply:
No needed. Reason being is if key3 is set, then keys 2 and 1 are set. All happening from the constructor that takes in 3 arguments.

Reply:
Thinking about this more, maybe I'll add checks for the other keys as well. Even though it's not likely that they are needed, it'll help guard against future code changes in case the keys are publicly exposed.

Reply:
sounds good.